packages/cli/src/MCP.ts

```diff
@@ -8,9 +8,29 @@ import { z } from 'zod'
 import { version } from '../package.json'
 import { createHelp } from './platform/_lib/help'
 
-async function runCommand({ args, cwd }: { args: string[]; cwd: string }) {
-  const result = await execa.node(process.argv[1], args, { cwd })
-  return `${result.stdout}\n${result.stderr}`
+// Only apply console redirection when running in MCP mode
+// This prevents stdout pollution that breaks MCP's JSON-RPC protocol
+if (process.argv.includes('mcp')) {
+  console.log = (...a: unknown[]) => process.stderr.write(a.map(String).join(' ') + '\n')
```

Comment:
This technically breaks much of the `console.log` functionality: this implementation will only work when printing the primitive values and not using any format specifiers. Should it be just this instead?
```suggestion
  console.log = console.error.bind(console)
```

Reply:
Also, where do those logs come from? Can we just print them to stderr in the first place if they're in our code instead of monkey-patching `console`?